sycl/source/detail/builtins.cpp

```diff
@@ -6,13 +6,18 @@
 //
 //===----------------------------------------------------------------------===//
 
+// Define _USE_MATH_DEFINES to enforce math defines of macros like M_PI.
+// Also, include this header first to avoid including it with unset
+// _USE_MATH_DEFINES via other sycl headers.
+#define _USE_MATH_DEFINES
+#include <cmath>
+#include <algorithm>
```

Comment:
I suggest moving standard C++ includes after SYCL headers includes to avoid accidental use of standard C++ headers w/o corresponding include directives in SYCL headers.

Reply:
I tried doing that in the updated commit, **BTW, disregard it, and do not approve it yet** as it does not work properly on Windows due to fact that <cmath> gets included somewhere in SYCL headers.

<cmath> is not included directly in SYCL header fiels (I checked that before updating the commit), but it is included indirectly/transitively via some standard/system header. That way leaves M_PI undeclared.

Even the following idea did not work. It looks like there is something in std headers that undefines _USE_MATH_DEFINES:

// Define _USE_MATH_DEFINES to enforce math defines of macros like M_PI in
// <cmath>. _USE_MATH_DEFINES is defined here before includes of SYCL header
// files to avoid include of <cmath> via those SYCL headers with unset
// _USE_MATH_DEFINES.
#define _USE_MATH_DEFINES

#include <CL/sycl/detail/generic_type_traits.hpp>
#include <CL/sycl/exception.hpp>
#include <CL/sycl/pointers.hpp>

#include <cmath>
#include <algorithm>

Reply:
Fixed. Thank you.